### PR TITLE
CIP8 name access via the CLI

### PR DESCRIPTION
**Pull request by AlexBHarley**
### Description

Improves upon our offchain-data commands in the CLI to allow writing / reading public / private names via the CLI.

### Other changes

Fixed the `register-data-encryption-key` command.

### Tested

Here's a script (literally a talking script in this case, not an executable file, it's going to be how I demo CIP8 in a tech talk) that you can follow to test these changes.

```
writer

mnemonic: tip young crunch caution during point noble pitch penalty mom clerk undo remember certain then mountain frequent region firm frozen device denial body cloud
privateKey: a9f4fba52c08e4ae8b9afc113acb8ffba6dab8704725a9f09112daec180fbd1d
publicKey: 02061e5d29df28d83929c067985d98de8e01780bc5352ec4c697a0996fbaa71f0c
address: 0x7ca840919dfdC593F6fC18cF12AC49827a5d2F41

writer dek

mnemonic: critic runway wasp swap oven various federal inherit ring lion remove lawn among hand illegal puppy shuffle step vast doctor shy floor lounge pyramid
privateKey: 3b39aa75f22e4cf46cb32702bde5b5a30a030d6ea523ff29f3ec7081b4666c0a
publicKey: 03223fa4abc2b01d9016a0986e7f1686e91ef031650e9b851bbc5ba856bcb2ffb1
address: 0xdc45d6aD3F8Ed82DBf01A9d2D1F0A6C6232887B8

reader

mnemonic: broom garbage tide curve shoulder mask symbol spoil pact forest gate another negative devote genius among error other broken congress adjust stadium post then
privateKey: 1e65bd2c33e96f4286549dc9effd2af16bf90a01a40c81bd11b50c4e0cfd183c
publicKey: 03a986914f87baf28da56ecef4c65391e3d27b85e6206820947396c3d35ae348a8
address: 0x227624fBA40E35Aa945a4E02bF2aD64eE1AB6b67

reader dek

mnemonic: stock apology social still twice because swim pink bargain burden anxiety over ticket rack mixed toast oven vanish brass silent lab admit hamster educate
privateKey: 86cf48c4f6e1e6d92a455fcd46302fac8317489e6b1f8000ae8c74bda95aeae3
publicKey: 033bfaf511d4818a400d4e9023957ab873ce6cd2172f9a1f5e0a404b84a1072dd9
address: 0xa68DCE7EbbC3B7f75a22650590E34be0b8576879


Pre steps:
1. Faucet writer address
2. Faucet reader address
3. Startup geth node 
  ~/celo/celo-blockchain/build/bin/geth --syncmode lightest --alfajores --rpc --rpcaddr 127.0.0.1 --rpcport 8545 --rpcapi eth,net,web3,debug,admin,personal,rpc --allow-insecure-unlock --datadir ./alfajores-lightest
4. Import keys and unlock accounts
  ~/celo/celo-blockchain/build/bin/geth attach http://localhost:8545
  personal.importRawKey('a9f4fba52c08e4ae8b9afc113acb8ffba6dab8704725a9f09112daec180fbd1d', '')
  personal.importRawKey('1e65bd2c33e96f4286549dc9effd2af16bf90a01a40c81bd11b50c4e0cfd183c', '')
  personal.unlockAccount('0x7ca840919dfdC593F6fC18cF12AC49827a5d2F41', '')
  personal.unlockAccount('0x227624fBA40E35Aa945a4E02bF2aD64eE1AB6b67', '')
5. Register writer
  ./packages/cli/bin/run account:register --from 0x7ca840919dfdC593F6fC18cF12AC49827a5d2F41
6. Register reader
  ./packages/cli/bin/run account:register --from 0x227624fBA40E35Aa945a4E02bF2aD64eE1AB6b67
7. Create metadata file 
  ./packages/cli/bin/run account:create-metadata ./metadata.json --from 0x7ca840919dfdC593F6fC18cF12AC49827a5d2F41
8. Claim storage
  ./packages/cli/bin/run account:claim-storage --from 0x7ca840919dfdC593F6fC18cF12AC49827a5d2F41 --url https://storage.googleapis.com/celo-test-alexh-bucket metadata.json
9. Copy to google storage 
  gsutil cp metadata.json gs://celo-test-alexh-bucket/metadata.json
10. Set metadata URL 
  ./packages/cli/bin/run account:register-metadata --url https://storage.googleapis.com/celo-test-alexh-bucket/metadata.json --from 0x7ca840919dfdC593F6fC18cF12AC49827a5d2F41
11. Register writer dek
  ./packages/cli/bin/run account:register-data-encryption-key --from 0x7ca840919dfdC593F6fC18cF12AC49827a5d2F41 --publicKey 03223fa4abc2b01d9016a0986e7f1686e91ef031650e9b851bbc5ba856bcb2ffb1
12. Register read dek
  ./packages/cli/bin/run account:register-data-encryption-key --from 0x227624fBA40E35Aa945a4E02bF2aD64eE1AB6b67 --publicKey 033bfaf511d4818a400d4e9023957ab873ce6cd2172f9a1f5e0a404b84a1072dd9

Navigate to https://console.cloud.google.com/storage/browser/celo-test-alexh-bucket;tab=objects?project=celo-testnet&prefix=&forceOnObjectsSortingFiltering=false

export SENDER=0x7ca840919dfdC593F6fC18cF12AC49827a5d2F41;
export SENDER_KEY=a9f4fba52c08e4ae8b9afc113acb8ffba6dab8704725a9f09112daec180fbd1d;
export SENDER_DEK=3b39aa75f22e4cf46cb32702bde5b5a30a030d6ea523ff29f3ec7081b4666c0a;

export READER_DEK=86cf48c4f6e1e6d92a455fcd46302fac8317489e6b1f8000ae8c74bda95aeae3;
export READER=0x227624fBA40E35Aa945a4E02bF2aD64eE1AB6b67;

Demo:
1. Private write
  cli account:offchain-write --privateDEK $SENDER_DEK --name alex --provider gcp --bucket celo-test-alexh-bucket --encryptTo $READER --privateKey $SENDER_KEY
2. Private read
  cli account:offchain-read $SENDER --privateDEK $READER_DEK --from $READER
3. Public read fails 
  cli account:offchain-read $SENDER --from $READER
4. Public write
  cli account:offchain-write --name alex --provider gcp --bucket celo-test-alexh-bucket --privateKey $SENDER_KEY
5. Public write succeeds
  cli account:offchain-read $SENDER
```

### CODE REVIEW

**Comment by nambrot** [comment]
Why not just add the keys to your local node to avoid passing the private keys to the CLI?

**Comment by AlexBHarley** [comment]
> Why not just add the keys to your local node to avoid passing the private keys to the CLI?

We need access to `eth_signTypedData` and our RPC API doesn't support that, only local wallets.

**Review by medhakothari — APPROVED** [review]
LGTM! Also, tested and it works.
However, I think the sdk docs can be removed (every file that begins with `developer-resources`). I believe only the one doc file starting with `command-line-interface` should be changed.

File: packages/cli/src/commands/account/register-data-encryption-key.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { ensureLeading0x } from '@celo/utils/src/address'
```
Comment (gastonponti) [inline packages/cli/src/commands/account/register-data-encryption-key.ts]: src -> lib

File: packages/cli/src/utils/off-chain-data.ts [hunk 2]
```diff
@@ -1,37 +1,60 @@
 import OffchainDataWrapper from '@celo/identity/lib/offchain-data-wrapper'
-import { GitStorageWriter, LocalStorageWriter } from '@celo/identity/lib/offchain/storage-writers'
+import {
+  AwsStorageWriter,
+  GitStorageWriter,
+  GoogleStorageWriter,
+  LocalStorageWriter,
+} from '@celo/identity/lib/offchain/storage-writers'
+import { privateKeyToAddress } from '@celo/utils/src/address'
```
Comment (gastonponti) [inline packages/cli/src/utils/off-chain-data.ts]: src -> lib

**Review by gastonponti — COMMENTED** [review]
Check these import changes

**Pull request merged**